app/src/main/java/ch/epfl/sdp/map/GoogleLocationFinder.java

```diff
@@ -29,10 +30,11 @@ public GoogleLocationFinder(LocationManager locationManager) {
         locationListener = new LocationListener() {
             @Override
             public void onLocationChanged(Location latestLocation) {
-                currentLocation = latestLocation;
-                PlayerManager.getInstance().getCurrentUser().setLocation(new GeoPoint(latestLocation.getLongitude(), latestLocation.getLatitude()));
-                PlayerManager.getInstance().getCurrentUser().setLocation(PlayerManager.getInstance().getCurrentUser().getLocation());
-                requestUpdatePosition();
+                if (Math.floor(latestLocation.getLatitude() * 100) / 100 != 37.42 || Math.ceil(latestLocation.getLongitude() * 100) / 100 != -122.08) {

```

Comment:
What is this formula?

Reply:
Sometimes, the player's location will change to this wired GeoPoint(which is the base of Google in the USA) automatically even the player stays in the fixed position. It seems like a problem from the emulator, so I filtered this location.

Reply:
Oh, I see